### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by Vikas-jangid**
Vuetify Form Validation Complete

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
+
     </v-main>
   </v-app>
 </template>
 <script>
-import Topnav from './components/Topnav.vue'
+import Topnav from "./components/Topnav.vue";
+
+
+
```
[the remark this answers is not in the document]
Comment (sunil4infobeans) [reply src/App.vue:15]: @Vikas-jangid pls remove these extra new lines.

File: src/App.vue [hunk 2]
```diff
@@ -2,12 +2,17 @@
   <v-app>
     <v-main>
       <top-nav></top-nav>
-     <router-view ></router-view>
+      <router-view></router-view>
+
+
```
Comment (sunil4infobeans) [inline src/App.vue:7]: Extra new lines need to be removed.

File: src/main.js [hunk 3]
```diff
@@ -6,8 +6,9 @@ import VueResource from 'vue-resource'
 import router from "./router/route";
 import Vuetify from 'vuetify/lib';
 
-// eslint-disable-next-line
 // prettier-disable-next-line
+/* eslint-disable */
+
```
Comment (sunil4infobeans) [inline src/main.js:13]: Pls remove non required commented code

File: src/views/AddEdit.vue [hunk 4]
```diff
@@ -5,66 +5,85 @@
...
+                  <v-text-field
+                    v-model="TodoData.date"
+                    label="Select Date"
+                    prepend-icon="mdi-calendar"
+                    readonly
+                    v-bind="attrs"
+                    v-on="on"
+                  ></v-text-field>
+                </template>
+                <v-date-picker v-model="TodoData.date" @input="menu2 = false"></v-date-picker>
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:50]: :+1:

File: src/views/AddEdit.vue [hunk 5]
```diff
@@ -73,10 +92,35 @@
 export default {
   data() {
     return {
-      date: new Date().toISOString().substr(0, 10),
+      menu2 : false,
+      TodoData: {
+        title: "",
+        priorities: ['High' , 'low'],
+        date: new Date().toISOString().substr(0, 10),
+        status: ['Completed' , 'About to Complete' , 'Pending'],
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:100]:
```suggestion
        status: ['Todo' , 'In Progress' , 'Done'],
```

File: src/views/AddEdit.vue [hunk 6]
```diff
@@ -73,10 +92,35 @@
...
+        description: "",
+      },
+      titleRule: [(v) => v.length != 0 || "Title is required"],
+      descriptionRule: [(v) => v.length > 10  || "Description should be about 10 characters"],
+      statusRule: [
+                    (v) => !!v || "Status is required",
+                    (v) =>  (v && v.length > 0) || "Status is required"
+                   ],
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:108]: Mark default stastus as "Todo", validation will not be required for this field.

File: src/views/AddEdit.vue [hunk 7]
```diff
@@ -73,10 +92,35 @@
 export default {
   data() {
     return {
-      date: new Date().toISOString().substr(0, 10),
+      menu2 : false,
+      TodoData: {
+        title: "",
+        priorities: ['High' , 'low'],
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:98]:
```suggestion
        priorities: ['Low', 'High', 'Very High'],
```

File: src/views/AddEdit.vue [hunk 8]
```diff
@@ -73,10 +92,35 @@
...
+      titleRule: [(v) => v.length != 0 || "Title is required"],
+      descriptionRule: [(v) => v.length > 10  || "Description should be about 10 characters"],
+      statusRule: [
+                    (v) => !!v || "Status is required",
+                    (v) =>  (v && v.length > 0) || "Status is required"
+                   ],
+      priorityRule: [
+                    (v) => !!v || "Status is required",
+                    (v) =>  (v && v.length > 0) || "Status is required"
+                   ],
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:112]: Mark default priority as "Low", validation will not be required for this field.

File: src/views/AddEdit.vue [hunk 9]
```diff
@@ -73,10 +92,35 @@
...
     };
   },
+  methods: {
+    SubmitForm() {
+      if (this.$refs.todoform.validate()) {
+       if(this.refs.TodoData.status == " "){
+         alert('status required');
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:119]: Use vuetify Dialogue.

File: src/views/AddEdit.vue [hunk 10]
```diff
@@ -73,10 +92,35 @@
...
+       }
+      }
+    },
+  },
 };
 </script>
-<style></style>
-
+<style></style>
```
Comment (sunil4infobeans) [inline src/views/AddEdit.vue:126]: Empty style statement is not required.

File: src/views/TodoList.vue [hunk 11]
```diff
@@ -137,5 +145,4 @@ export default {
   }
 };
 </script>
-<style></style>
-    
+<style></style>
```
Comment (sunil4infobeans) [inline src/views/TodoList.vue:148]: Empty style statement is not required.

**Review by sunil4infobeans — CHANGES_REQUESTED** [review]
Requires a few updates.

File: src/views/AddEdit.vue (hunk 7 above)
Comment (Vikas-jangid) [reply src/views/AddEdit.vue:98]: Priorities Changed As per  recommendation

File: src/views/AddEdit.vue (hunk 5 above)
Comment (Vikas-jangid) [reply src/views/AddEdit.vue:100]: Status Options are updated

File: src/views/AddEdit.vue (hunk 9 above)
Comment (Vikas-jangid) [reply src/views/AddEdit.vue:119]: Ok

File: src/views/TodoList.vue (hunk 11 above)
Comment (Vikas-jangid) [reply src/views/TodoList.vue:148]: ok

**Comment by sunil4infobeans** [comment]
Closing this PR against https://github.com/Vikas-jangid/todo-app/pull/5

**Pull request closed without merging**